WordPress/gutenberg
# Writing flow: Try arrow press confirmation before switching blocks

## What?


The PR changes writing flow to require a confirmation arrow key press after reaching the border of a block to switch blocks.

## Why?


This has been a long time hated feature for me because as a screen reader user, it is not always apparent when I am at the edge of a block and my next arrow press sends me into another block with no warning. From that, a setting was added to restrict this behavior. Eventually, Slack Accessibility team tried something new. If the Slack composer field for a DM or channel contains text, you can navigate to the top of the field on Windows by pressing up arrow or ctrl+home. However, once you finally reach the top of the field, you must press up arrow another time to confirm that you wish to leave the compose field and enter the DM/channel message list. I figured, why not start a draft PR and play around with it for the block editor. It was not hard to get an initial POC, all I need now is the community to be a little open minded as this could make the experience better for everyone.

It is worth saying that this might still not be enough in the end to make the experience as unified for screen reader users as it is for the sighted, but maybe it could be a good first step.

## How?


Essentially, I am just tracking this within a variable that is true or false depending on certain conditions. This is all hooked up in `useArrowNav()` writing flow hook.

## Testing Instructions





1. Open a post or page.
2. Insert some blocks with different text.
3. In the last block, press arrow up twice to move to the second block.
4. Test moving down, left, right, and up. You should be required to press the arrow key twice as a confirmation action before the switch happens.

### Testing Instructions for Keyboard


## Screenshots or screencast 

## To-do:
1. [ ] Fix the E2E tests.
2. [x] Currently, the local variable is not set to false if you move away from the vertical or horizontal edge. Trying to figure out how to solve that.
3. [ ] I need a way to check if a block is empty which I guess really only leaves us with a default block?
